projects/core/src/lib/layer-collections/layer-collection.service.ts

```diff
@@ -45,4 +76,186 @@ export class LayerCollectionService {
             .getSessionStream()
             .pipe(mergeMap((session) => this.backend.getWorkflowMetadata(workflowId, session.sessionToken)));
     }
+
+    /**
+     * Add all layers (directly) contained in a layer collection to the current project.
+     */
+    addCollectionLayersToProject(collection: LayerCollectionDict): void {
+        const layersObservable = collection.items
+            .filter((layer) => layer.type === 'layer')
+            .map((layer) => layer as LayerCollectionLayerDict)
```

Comment:
Do you get the "wrong" type here? You basically just typecast-map, right?

Reply:
a collection can also contain child collections. This method only adds the directly contained layers, that's why I filter here.

Reply:
I don't mean the filter, but the `map`.